File: vue-crud-app/src/services/TypeTransactionService.js
import http from "../http-common";

class TypeTransactionService {
  getAll() {
    return http.get("/type-transaction");
  }

  get(id) {
    return http.get(`/type-transaction/${id}`);
  }

  create(data) {
    return http.post("/type-transaction", data);
  }

  update(id, data) {
    return http.put(`/type-transaction/${id}`, data);
  }

  delete(id) {
    return http.delete(`/type-transaction/${id}`);
  }

  deleteAll() {
    return http.delete(`/type-transaction`);
  }

  findByTitle(title) {
    return http.get(`/type-transaction?title=${title}`);
  }
}

export default new type-transaction();

File: Bank-backend/src/main/java/com/ipnet/Bank_backend/services/TypeTransactionService.java
package com.ipnet.Bank_backend.services;

import com.ipnet.Bank_backend.entities.TypeTransaction;

import java.util.List;

public interface TypeTransactionService {
    TypeTransaction create (TypeTransaction typeTransaction);
    TypeTransaction update (TypeTransaction typeTransaction, Long id);
    List<TypeTransaction> getAll();
    void delete (Long id);
    TypeTransaction findById (Long id);
}
